config-scripts/cups-common.m4

```diff
@@ -34,13 +34,18 @@ CPPFLAGS="${CPPFLAGS:=}"
 CXXFLAGS="${CXXFLAGS:=}"
 LDFLAGS="${LDFLAGS:=}"
 
+dnl Setting compiler environment
+AR="${AR:=}"
+
```

Comment:
Autoconf's generated configure script already allows the environment variable to override the configure check.

Reply:
I tried to pass these environment variables in 'configure' script without my changes and 'ar' is obtained from the system by searching for 'ar' program using PATH:
```
AR=llvm-ar CC=clang CXX=clang++ CFLAGS=-flto CXXFLAGS=-flto LDFLAGS=-flto ./configure
```
at the end of the build I see
```
*/usr/bin/ar* crvs libcups.a adminutil.o array.o auth.o backchannel.o backend.o debug.o dest.o dest-job.o dest-localization.o dest-options.o dir.o encode.o file.o getdevices.o getifaddrs.o getput
file.o globals.o hash.o http.o http-addr.o http-addrlist.o http-support.o ipp.o ipp-support.o langprintf.o language.o md5.o md5passwd.o notify.o options.o ppd.o ppd-attr.o ppd-cache.o ppd-conflicts.o ppd-custom.o ppd-emit.o ppd-localize.o ppd-mark.o ppd-page.o ppd-util.o pwg-media.o request.o sidechannel.o snmp.o snprintf.o string.o tempfile.o thread.o tls.o transcode.o usersys.o util.o
```

[here](https://www.gnu.org/software/autoconf/manual/autoconf-2.68/html_node/Generic-Programs.html) manual says, quote, 
> Macro: AC_PATH_PROGS (variable, progs-to-check-for, [value-if-not-found], [path = ‘$PATH’])
> > Like AC_CHECK_PROGS, but if any of progs-to-check-for are found, set variable to the absolute name of the program found. *The result of this test can be overridden by setting the _variable_ variable*. A positive result of this test is cached in the ac_cv_path_variable variable.

I think it does not mean that the result can be ignored in a favor of an environment variable with the same name, does it?